crates/data_components/src/graphql/provider.rs

```diff
@@ -27,19 +27,39 @@ use datafusion::{
 };
 use std::{any::Any, sync::Arc};
 
-use super::client::GraphQLClient;
 use super::Result;
+use super::{client::GraphQLClient, ResultTransformSnafu};
+
+pub type TransformFn = Arc<
+    dyn Fn(&RecordBatch) -> Result<RecordBatch, Box<dyn std::error::Error + Send + Sync>>
```

Comment:
Can this just be a function pointer instead? https://doc.rust-lang.org/reference/types/function-pointer.html

It doesn't need to be a closure since the function itself doesn't close over any environment variables.

Reply:
Here is an example of using a function pointer in Spice: https://github.com/spiceai/spiceai/blob/trunk/crates/data_components/src/unity_catalog/provider.rs#L41

Reply:
Changed to function pointer